src/nxdoc/nuxeo-server/administration/observability/logs.md

```diff
@@ -0,0 +1,232 @@
+
+1.  Add the following line in your `nuxeo.conf`:
+
+    *   For VCS:
+
+        ```
+        org.nuxeo.vcs.query.log_min_duration_ms=100
+
+        ```
+
+    *   For DBS:
+
+        ```
+        org.nuxeo.dbs.query.log_min_duration_ms=100
+
+        ```
+
+2.  Activate the log for session you use (VCS or DBS)&nbsp;at the INFO level.
+    Here an example of `lib/log4j2.xml` to get them in a separate file:
+
+    *   For VCS:
+
+        ```xml
+        <category name="org.nuxeo.ecm.core.storage.sql.SessionImpl" additivity="false">
+            <priority value="INFO" />
+            <appender-ref ref="SLOW" />
+        </category>
+        <appender name="SLOW" class="org.apache.log4j.FileAppender">
+            <errorHandler class="org.apache.log4j.helpers.OnlyOnceErrorHandler" />
+            <param name="File" value="${nuxeo.log.dir}/slow-query.log" />
+            <param name="Append" value="false" />
+            <layout class="org.apache.log4j.PatternLayout">
+              <param name="ConversionPattern" value="%d{ISO8601} %-5p [%t][%c] %m%X%n" />
+            </layout>
+        </appender>
+        ```
+
+    *   For DBS:
+
+        ```xml
+        <category name="org.nuxeo.ecm.core.storage.dbs.DBSSession" additivity="false">
+            <priority value="INFO" />
+            <appender-ref ref="SLOW" />
+        </category>
+        <appender name="SLOW" class="org.apache.log4j.FileAppender">
+            <errorHandler class="org.apache.log4j.helpers.OnlyOnceErrorHandler" />
+            <param name="File" value="${nuxeo.log.dir}/slow-query.log" />
+            <param name="Append" value="false" />
+            <layout class="org.apache.log4j.PatternLayout">
+              <param name="ConversionPattern" value="%d{ISO8601} %-5p [%t][%c] %m%X%n" />
+            </layout>
+        </appender>
+        ```
+
+To get the slowest queries:
+
+```bash
+$ grep duration_ms log/slow-query.log | sed -e's/^[^\t]*\t//g' -e 's/{.*$//g' |sort -nr | head
+
+802.89  QueryFilter(principal=system, limit=0, offset=0)        query   SELECT * FROM Document WHERE ....
+12.87   QueryFilter(principal=bob, limit=0, offset=0)        queryAndFetch   Select DISTINCT ecm:uuid...
+10.12   QueryFilter(principal=system, limit=50, offset=0) count total results UNLIMITED query   SELECT * FROM Document WHERE ...
+5.35    QueryFilter(principal=Administrator, limit=20, offset=0) count total results up to 20   query   SELECT * FROM ...
+
+```
+
+{{#> callout type='tip' }}
+
+When configuring the log priority to TRACE a debug stack trace is available in the log so you got the code path.
+
+{{/callout}}
+
+## Logging Transactional Events
+
+Configure log4j in `$NUXEO_HOME/lib/log4j2.xml` by adding the following keywords to your appender conversion pattern `%t` for the thread name and `%X`&nbsp;for the logging context map:
+
+```html/xml
+<param name="ConversionPattern" value="%d{ISO8601} %t %-5p [%c] %m %X%n" />
+```
```

Comment:
Maybe something like below? I checked our appenders, almost all of them have the `%t` keywords.
```suggestion
Configure log4j in `$NUXEO_HOME/lib/log4j2.xml` by adding the following keywords to your appender pattern layout: `%t`&nbsp;for the thread name and `%X`&nbsp;for the logging context map:

```html/xml
<PatternLayout pattern="%d{ISO8601} %-5p [%t] [%c] %m%X%n" />
```
```